api/controllers/game/rematch-gamestate.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Endpoint to request/decline playing again
+ * If both players accept rematch, creates
+ * a new game, switching who goes first
+ * 
+ * New game will be ranked/casual based on previous match
+ * with name "firstPlayerUsername VS secondPlayerUsername {p0wins}-{p1Wins}-{stalemates}"
+ */
+const gameAPI = sails.hooks['customgamehook'];
+module.exports = async function (req, res) {
+  try {
+    const { usr: userId } = req.session;
+    const { gameId: oldGameId } = req.params;
+    const { rematch } = req.body;
+
+    let game = await sails.helpers.lockGame(req.session.game);
+
+    // Early return if requesting user was not in the game
+    if (!userId || ![ game.p0?.id, game.p1?.id ].includes(userId)) {
+      return;
+    }
```

Comment:
Let's handle the unauthenticated case (`!userId`) by applying the `isLoggedIn` policy to this endpoint in `config/policies.js`. Then let's update this to specifically return a 403 response to signify that if you're not in the game, you're not allowed to do that

```suggestion
    if (![ game.p0?.id, game.p1?.id ].includes(userId)) {
      return res.forbidden({ message: `You aren't a player in this game` });
    }
```

This also reminds me that our `isLoggedIn` policy itself incorrectly throws a 403 (using `res.forbidden()` when it should throw 401 for unauthenticated. I'll write an issue for that follow up